### PR TITLE
stream prepare has nullary overload

### PR DESCRIPTION
**Pull request by cmazakas**

### CODE REVIEW

**Comment by vinniefalco** [comment]
Well shouldn't we remove the 1-arg overload?

File: include/boost/http_proto/serializer.hpp [hunk 1]
```diff
@@ -247,25 +247,35 @@ class BOOST_SYMBOL_VISIBLE
...
-    last_chunk_len_ = 1 + 2 + 2; // chunked-body termination requires an extra CRLF
+    last_chunk_len_ =
+        1 + // "0"
+        crlf_len_ +
+        crlf_len_; // chunked-body termination requires an extra CRLF
 
     static
     constexpr
     std::size_t
     chunked_overhead_ =
```
Comment (ashtum) [inline include/boost/http_proto/serializer.hpp:276]: We might be able to use this constant in serializer::prepare() and serializer::capacity() too.

**Comment by cmazakas** [comment]
> Well shouldn't we remove the 1-arg overload?

Good question. I didn't see a reason to outright remove it.

I suppose users can take the return from `stream::preapre()` and then use Buffers to manipulate it. If you want, I can go ahead and remove the `stream::preapre(std::size_t);` overload

File: src/serializer.cpp [hunk 2]
```diff
@@ -560,14 +560,29 @@ size() const
...
 {
+    auto n = sr_->tmp0_.capacity();
     if( sr_->is_chunked_ )
+    {
+        // for chunked encoding, we want to unconditionally
+        // reserve space for the complete chunk and the
+        // last-chunk
+        // this enables users to call:
+        //
+        //     stream.commit(n); stream.close();
```
Comment (vinniefalco) [inline src/serializer.cpp:574]: Nice comment... yes, this is a good goal

**Pull request merged**